src/ioctls/system.rs

```diff
@@ -228,13 +228,18 @@ impl Kvm {
     }
 
     #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
-    fn get_cpuid(&self, kind: u64, max_entries_count: usize) -> Result<CpuId> {
-        let mut cpuid = CpuId::new(max_entries_count);
+    fn get_cpuid(&self, kind: u64, num_entries: usize) -> Result<CpuId> {
+        if num_entries > KVM_MAX_CPUID_ENTRIES {
+            // Returns the same error the underlying `ioctl` would have sent.
+            return Err(errno::Error::new(ENOMEM));
```

Comment:
nit: Actually newer kernels return `E2BIG`, you could do the same.

Reply:
Actually `E2BIG` is returned if the passed number of entries is too low and `ENOMEM` is returned if the passed number of entries is too high. So I believe in this case `ENOMEM` is right - as per [this documentation](https://www.kernel.org/doc/html/latest/virt/kvm/api.html?highlight=kvm%20ioctl)